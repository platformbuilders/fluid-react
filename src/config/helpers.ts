import { DefaultTheme } from 'styled-components';
import { animations, colors, radius, spacing } from '../theme';

export const theme: DefaultTheme = {
  ...colors,
  ...animations,
  ...radius,
  ...spacing,
};
