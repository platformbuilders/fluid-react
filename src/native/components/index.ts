// Inputs
export { default as FormError } from './FormError';
export { default as TextInput } from './TextInput';
export { default as PinInput } from './PinInput';
export { default as PasswordInput } from './TextInput/PasswordInput';
// Buttons
export { default as Touchable } from './Touchable';
export { default as Icon } from './Icon';
export { default as Button } from './Button';
// General
export { default as If } from './If';
export { default as Typography } from './Typography';
export { default as CheckBox } from './CheckBox';
export { default as LoadingIndicator } from './LoadingIndicator';
export { default as Shadow } from './Shadow';
export { default as DatePicker } from './DatePicker';
export { default as Avatar } from './Avatar';
export { default as Link } from './Link';
export { default as Accordion } from './Accordion';
export { default as RadioButton } from './RadioButton';
