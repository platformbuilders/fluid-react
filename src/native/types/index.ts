export * from './Accordion';
export * from './Common';
export * from './Colors';
export * from './ThemeType';
export * from './Metrics';
export * from './Radius';
export * from './Variants';
export * from './TypographyType';
export * from './InputStatus';
export * from './TouchableType';
export * from './IconType';
export * from './TextInputType';
export * from './Animations';
export * from './Loading';
