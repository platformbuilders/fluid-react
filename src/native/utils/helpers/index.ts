export * from '@platformbuilders/helpers';
export * from './generateHaptic';
