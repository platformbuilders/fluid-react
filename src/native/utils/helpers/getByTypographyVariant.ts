/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import { getTheme } from '@platformbuilders/helpers';
import { scaledFontSize } from './fontScale';

export const getFontSize = (props: any): any =>
  scaledFontSize(getTheme(`${props.variant}.fontSize`)(props));

export const getLineHeight = (props: any): any =>
  scaledFontSize(getTheme(`${props.variant}.lineHeight`)(props));
