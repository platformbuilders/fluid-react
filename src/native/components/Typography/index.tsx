import React, { FC } from 'react';
import { human } from 'react-native-typography';
import { TypographyType } from '../../types';
import { Text } from './styles';

const Typography: FC<TypographyType> = ({
  style = [{}],
  variant = 'body',
  textRef = React.createRef(),
  children,
  id,
  accessibility,
  ...rest
}) => (
  <Text
    testID={id}
    accessibilityLabel={accessibility}
    ref={textRef}
    style={[human[variant], style]}
    {...rest}
  >
    {children}
  </Text>
);

export default Typography;
