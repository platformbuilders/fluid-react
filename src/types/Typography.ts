import { PropsWithChildren } from 'react';
import { KnownTarget } from 'styled-components/dist/types';
import { TypographyVariants } from '@platformbuilders/theme-toolkit';

type TypographyTypeBase = PropsWithChildren<{
  variant?: TypographyVariants;
  id?: string;
  accessibility?: string;
  numberOfLines?: number;
  as?: KnownTarget;
}>;

export type TypographyTypeProps = TypographyTypeBase & {
  lineHeightVariant?: TypographyVariants;
};

export type TypographyTypeStyleProps = TypographyTypeBase & {
  $lineHeightVariant?: TypographyVariants;
  $variant?: TypographyVariants;
};
