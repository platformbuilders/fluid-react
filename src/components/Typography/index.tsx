import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { human } from 'react-native-typography';
import { Theme, Metrics, Variant } from '~/utils/types';
import { Text } from './styles';
import { ThemeContext } from '../ThemeContext';

interface Props {
  variant: Variant;
  children?: string | string[];
  style?: object[];
  textRef?: any;
  theme: Theme & Metrics;
}

const Typography: React.FC<Props> = ({
  style = [{}],
  variant = 'body',
  textRef = React.createRef(),
  children,
  ...rest
}) => {
  const { theme } = useContext(ThemeContext);
  return (
    <ThemeProvider theme={theme}>
      <Text
        ref={textRef}
        style={[human[variant], { color: theme.primary.contrast }, style]}
        {...rest}
      >
        {children}
      </Text>
    </ThemeProvider>
  );
};

export default Typography;
