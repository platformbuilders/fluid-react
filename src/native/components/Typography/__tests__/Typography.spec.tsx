import 'jest';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import Typography from '..';
import { theme } from '../../../test/helpers';

describe('<Typography />', () => {
  it('should render and match snapshot for default props', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant largeTitle', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="largeTitle" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title1', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="title1" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title2', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="title2" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title3', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="title3" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant title4', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="title4" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant headline', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="headline" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant subhead', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="subhead" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant footnote', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="footnote" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant caption1', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="caption1" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render and match snapshot for variant caption2', () => {
    const render = renderer.create(
      <ThemeProvider theme={theme}>
        <Typography variant="caption2" />
      </ThemeProvider>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
