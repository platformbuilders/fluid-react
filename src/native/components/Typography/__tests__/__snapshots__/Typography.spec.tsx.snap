// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should render and match snapshot for default props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "400",
        "letterSpacing": -0.408,
        "lineHeight": 22,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant caption1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 12,
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 16,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant caption2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 11,
        "fontWeight": "400",
        "letterSpacing": 0.06,
        "lineHeight": 13,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant footnote 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 13,
        "fontWeight": "400",
        "letterSpacing": -0.078,
        "lineHeight": 18,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant headline 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 17,
        "fontWeight": "600",
        "letterSpacing": -0.408,
        "lineHeight": 22,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant largeTitle 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 34,
        "fontWeight": "400",
        "letterSpacing": 0.381836,
        "lineHeight": 41,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant subhead 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 15,
        "fontWeight": "400",
        "letterSpacing": -0.24,
        "lineHeight": 20,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant title1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 28,
        "fontWeight": "400",
        "letterSpacing": 0.355469,
        "lineHeight": 34,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant title2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 22,
        "fontWeight": "400",
        "letterSpacing": 0.34375,
        "lineHeight": 28,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant title3 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      Object {
        "backgroundColor": "transparent",
        "color": "#000000",
        "fontFamily": "System",
        "fontSize": 20,
        "fontWeight": "400",
        "letterSpacing": 0.361328,
        "lineHeight": 25,
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;

exports[`<Typography /> should render and match snapshot for variant title4 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#ffffff",
      },
      undefined,
      Array [
        Object {},
      ],
    ]
  }
/>
`;
